test/gtest/uct/test_md.cc

```diff
@@ -310,13 +310,15 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
             offset        = ucs::rand() % buffer_size;
             length_offset = ucs::rand() % buffer_size;
 
-            status = uct_md_mem_query(md(), (void*)((char*)address + offset),
+            status = uct_md_mem_query(md(),
+                                      UCS_PTR_BYTE_OFFSET(address, offset),
                                       length_offset, &mem_attr);
             ASSERT_UCS_OK(status);
             EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
-            EXPECT_TRUE((mem_attr.alloc_length >= buffer_size) ||
-                        (mem_attr.alloc_length == -1));
-            EXPECT_TRUE((mem_attr.base_address == NULL) ||
+            EXPECT_TRUE((mem_attr.alloc_length == buffer_size) ||
+                        (mem_attr.alloc_length == length_offset));
+            EXPECT_TRUE((mem_attr.base_address ==
+                         UCS_PTR_BYTE_OFFSET(address, offset)) ||
```

Comment:
if mem type is cuda, can we expect `mem_attr.base_address == UCS_PTR_BYTE_OFFSET(address, offset)` ?

Reply:
@yosefe I don't think so ([Refer](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g64fee5711274a2a0573a789c94d8299b)). The second part of this condition check works for cuda. You would see base_address set to the start of the region allocated by cuda.

Reply:
right, so in case of cuda mem_attr.base_address==address? can we check this explicitly?

Reply:
Sure. I can add that check.

Reply:
@yosefe I reverted the above change because to handle cases where `cuMemGetAddressRange` fails. In this case, I've changed logic in uct cuda_copy to also return provided address and length. I hope that's ok.